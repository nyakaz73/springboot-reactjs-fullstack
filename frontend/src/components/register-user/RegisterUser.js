import React, { useState } from 'react';
import InputRow from '../row/InputRow';
import Button from "../button/Button";
import PropTypes from "prop-types";

const RegisterUser = ({ addUser }) => {
    const [user, setUser] = useState({
        name: '',
        surname: '',
        email: '',
        username: '',
        password: '',
        confirm_password: ''
    });

    const handleInputChange = (e) => {
        const { name, value } = e.target;
        setUser({
            ...user,
            [name]: value
        })
    }

    const onSubmit = (e) => {
        e.preventDefault();
        if (user['password'] === user['confirm_password']) {
            addUser(user);
            setUser({
                'name': '',
                'surname': '',
                'email': '',
                'username': '',
                'password': '',
                'confirm_password': ''
            })
        } else {
            console.log("Incorrect Password !")
        }


    }

    return (
        <div className="formCard">
            <form onSubmit={onSubmit} >
                <div className="pill"></div>
                <br />
                <div style={row}>
                    <InputRow label="Name">
                        <input
                            type="text"
                            name="name"
                            placeholder="Name"
                            value={user['name']}
                            onChange={handleInputChange}
                        />
                    </InputRow>
                    <InputRow label="Surname">
                        <input
                            type="text"
                            name="surname"
                            placeholder="Surname"
                            value={user['surname']}
                            onChange={handleInputChange}
                        />
                    </InputRow>
                </div>

                <div style={row}>
                    <InputRow label="Email">
                        <input
                            type="email"
                            name="email"
                            placeholder="Email"
                            value={user['email']}
                            onChange={handleInputChange}
                        />
                    </InputRow>
                    <InputRow label="Username">
                        <input
                            type="text"
                            name="username"
                            placeholder="Username"
                            value={user['username']}
                            onChange={handleInputChange}
                        />
                    </InputRow>
                </div>

                <div style={row}>
                    <InputRow label="Password">
                        <input
                            type="password"
                            name="password"
                            placeholder="Password"
                            value={user['password']}
                            onChange={handleInputChange}
                        />
                    </InputRow>
                    <InputRow label="Confirm Password">
                        <input
                            type="password"
                            name="confirm_password"
                            placeholder="Confirm Password"
                            value={user['confirm_password']}
                            onChange={handleInputChange}
                        />
                    </InputRow>

                </div>
                <div style={button}>
                    <Button />
                </div>

            </form>

        </div>
    )
}


const row = {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'center'
}

const button = {
    display: 'flex',
    justifyContent: 'flex-end',
    padding: '10px'

}

RegisterUser.propTyoes = {
    addUser: PropTypes.func.isRequired,
}

export default RegisterUser;
