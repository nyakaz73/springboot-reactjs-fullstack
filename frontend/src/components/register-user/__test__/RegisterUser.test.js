import React from "react";
import ReactDOM from "react-dom";
import { act, fireEvent, render } from "@testing-library/react";
import RegisterUser from "../RegisterUser";
import renderer from "react-test-renderer"

let container;

beforeEach(() => {
    container = document.createElement('div');
    document.body.appendChild(container);
});

afterEach(() => {
    document.body.removeChild(container);
    container = null;
});

it("renders RegisterUser without crashing", () => {
    act(() => {
        ReactDOM.render(<RegisterUser />, container)
    })

})

const setup = () => {

}
//To test the state of the form we use Enzyme simulate or React Testing Lib simulate events
it("update on onChange state on input form", () => {
    const utils = render(<RegisterUser />);
    const input = utils.getByPlaceholderText('Name');
    fireEvent.change(input, { target: { value: 'John' } })
    expect(input.value).toBe("John")
})

