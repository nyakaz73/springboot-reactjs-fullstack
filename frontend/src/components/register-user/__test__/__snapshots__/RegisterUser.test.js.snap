// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshots 1`] = `
<div
  className="formCard"
>
  <form
    onSubmit={[Function]}
  >
    <div
      className="pill"
    />
    <br />
    <div
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <div
        style={
          Object {
            "display": "flex",
            "flexDirection": "column",
            "padding": "10px",
          }
        }
      >
        <p
          style={
            Object {
              "color": "#1010108a",
              "fontWeight": "500",
              "paddingLeft": "16px",
            }
          }
        >
          Name
        </p>
        <div
          className="styledInput"
        >
          <input
            name="name"
            onChange={[Function]}
            placeholder="Name"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        style={
          Object {
            "display": "flex",
            "flexDirection": "column",
            "padding": "10px",
          }
        }
      >
        <p
          style={
            Object {
              "color": "#1010108a",
              "fontWeight": "500",
              "paddingLeft": "16px",
            }
          }
        >
          Surname
        </p>
        <div
          className="styledInput"
        >
          <input
            name="surname"
            onChange={[Function]}
            placeholder="Surname"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <div
        style={
          Object {
            "display": "flex",
            "flexDirection": "column",
            "padding": "10px",
          }
        }
      >
        <p
          style={
            Object {
              "color": "#1010108a",
              "fontWeight": "500",
              "paddingLeft": "16px",
            }
          }
        >
          Email
        </p>
        <div
          className="styledInput"
        >
          <input
            name="email"
            onChange={[Function]}
            placeholder="Email"
            type="email"
            value=""
          />
        </div>
      </div>
      <div
        style={
          Object {
            "display": "flex",
            "flexDirection": "column",
            "padding": "10px",
          }
        }
      >
        <p
          style={
            Object {
              "color": "#1010108a",
              "fontWeight": "500",
              "paddingLeft": "16px",
            }
          }
        >
          Username
        </p>
        <div
          className="styledInput"
        >
          <input
            name="username"
            onChange={[Function]}
            placeholder="Username"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <div
        style={
          Object {
            "display": "flex",
            "flexDirection": "column",
            "padding": "10px",
          }
        }
      >
        <p
          style={
            Object {
              "color": "#1010108a",
              "fontWeight": "500",
              "paddingLeft": "16px",
            }
          }
        >
          Password
        </p>
        <div
          className="styledInput"
        >
          <input
            name="password"
            onChange={[Function]}
            placeholder="Password"
            type="password"
            value=""
          />
        </div>
      </div>
      <div
        style={
          Object {
            "display": "flex",
            "flexDirection": "column",
            "padding": "10px",
          }
        }
      >
        <p
          style={
            Object {
              "color": "#1010108a",
              "fontWeight": "500",
              "paddingLeft": "16px",
            }
          }
        >
          Confirm Password
        </p>
        <div
          className="styledInput"
        >
          <input
            name="confirm_password"
            onChange={[Function]}
            placeholder="Confirm Password"
            type="password"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      style={
        Object {
          "display": "flex",
          "justifyContent": "flex-end",
          "padding": "10px",
        }
      }
    >
      <button
        className="btn"
      >
        Submit
      </button>
    </div>
  </form>
</div>
`;
